### PR TITLE
[CT-3230] Credit card support

### PR DESCRIPTION
**Pull request by vassildinev**
https://payhawk.atlassian.net/browse/CT-3220

- same logic as with Debit accounts
- we create credit accounts
- we send bills and deposits based on payment balance type

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ require('source-map-support').install();
     const integrationsController = Controllers.createIntegrations();
     const server = createServer(authController, integrationsController);
 
-    server.post('/migrate', async (req, res) => {
+    server.post('/events/schema-migrate', async (req, res) => {
```
Comment (StanislavRadkov) [inline src/index.ts:28]: Why was this modified? How did this worked before?

Comment (vassildinev) [reply src/index.ts:28]: since the new cluster we have not deployed a migration in Xero, hence why this change is now requred

File: src/managers/integration/Manager.spec.ts [hunk 2]
```diff
@@ -178,14 +180,14 @@ describe('integrations/Manager', () => {
...
 
-            payhawkClientMock
-                .setup(c => c.synchronizeBankAccounts([{
-                    externalId: businessBankAccount.accountID!,
-                    name: businessBankAccount.name!,
-                    number: businessBankAccount.bankAccountNumber!,
-                    currency: businessBankAccount.currencyCode!.toString(),
-                }]))
-                .verifiable(TypeMoq.Times.once());
+            // payhawkClientMock
```
Comment (StanislavRadkov) [inline src/managers/integration/Manager.spec.ts:183]: Why is this commented?

File: src/managers/integration/Manager.spec.ts [hunk 3]
```diff
@@ -328,8 +341,8 @@ describe('integrations/Manager', () => {
                             amount: settledBalancePayment.amount,
                             bankAccountId,
                             currency: settledBalancePayment.currency,
-                            date: settledBalancePayment.date,
-                            bankFees: settledBalancePayment.fees,
+                            date: settledBalancePayment.date.toISOString(),
```
Comment (StanislavRadkov) [inline src/managers/integration/Manager.spec.ts:344]: Why the call to `toISOString` ?

File: src/managers/integration/Manager.ts [hunk 4]
```diff
@@ -166,27 +166,17 @@ export class Manager implements IManager {
...
-        const bankAccountModels = bankAccounts
-            .filter(b => !payhawkBankAccountIds.has(b.accountID))
-            .map(b => ({
-                name: b.name,
-                externalId: b.accountID,
-                number: b.bankAccountNumber,
-                currency: b.currencyCode.toString(),
-            }));
-
-        await this.payhawkClient.synchronizeBankAccounts(bankAccountModels);
```
Comment (StanislavRadkov) [inline src/managers/integration/Manager.ts:187]: Why is this removed? Are we using the data from the `business_accounts` table?

File: src/managers/integration/Manager.spec.ts (hunk 2 above)
Comment (StanislavRadkov) [reply src/managers/integration/Manager.spec.ts:183]: This is legacy functionality

File: src/managers/integration/Manager.spec.ts (hunk 3 above)
Comment (StanislavRadkov) [reply src/managers/integration/Manager.spec.ts:344]: For consistency

**Review by StanislavRadkov — APPROVED** [review]
Looks good 👍

File: src/managers/integration/Manager.ts (hunk 4 above)
Comment (StanislavRadkov) [reply src/managers/integration/Manager.ts:187]: The `business_accounts` functionality is legacy and is no longer needed
